src/reader.c

```diff
@@ -107,7 +107,19 @@ static PyObject *createDecodedString(hiredis_ReaderObject *self, const char *str
 static void *createError(PyObject *errorCallable, char *errstr, size_t len) {
     PyObject *obj;
 
+    /* "s" stands for "string" (string object, byte string) in Python 2
+     * and "str" (str object, unicode string) in Python 3 */
     PyObject *args = Py_BuildValue("(s#)", errstr, len);
+    if (args == NULL) {
+        #if IS_PY3K
+        /* Clear an error indicator and construct bytes object ("y") instead of str
+         * in case of UnicodeDecodeError (e.g., malformed utf-8 string) in Python 3 */
+        if (PyErr_ExceptionMatches(PyExc_UnicodeDecodeError)) {
+            PyErr_Clear();
+            args = Py_BuildValue("(y#)", errstr, len);
```

Comment:
Is this a good idea? Behaviors are inconsistent. But I can't find a better solution.
What do you think about raising UnicodeDecodeError directly?

Reply:
I agree that this is an inconsistent behavior and this requires additional type check by the caller, but I think that `UnicodeDecodeError` is an even more unexpected behavior. Maybe decoding bytes to a string object in `replace` or `backslashreplace` mode would be a better solution?

Reply:
Decoding to string with `replace` mode on would be better, will work just like redis-cli.

Reply:
I've just opened new PR with suggested solution https://github.com/redis/hiredis-py/pull/73.